identity_bench_test.go

```diff
@@ -0,0 +1,40 @@
+package identity_test
+
+import (
+	"github.com/TankerHQ/identity-go/v3"
+	"testing"
+)
+
+func BenchmarkCreate(b *testing.B) {
+	for i := 0 ; i < b.N ; i++ {
+		identity.Create(validConf, "userID")
+	}
+}
+
+func BenchmarkCreateProvisional(b *testing.B) {
+	for _, target := range validTargets {
+		b.Run(target, func(b *testing.B) {
+			for i := 0; i < b.N; i++ {
+				identity.CreateProvisional(validConf, target, "userID")
+			}
+		})
+	}
+}
+
+func BenchmarkGetPublicIdentity(b *testing.B) {
+	for _, target := range validTargets {
+		provIdentity, _ := identity.CreateProvisional(validConf, target, "userID")
+		b.Run(target, func(b *testing.B) {
+			for i := 0; i < b.N; i++ {
+				identity.GetPublicIdentity(*provIdentity)
+			}
+		})
+	}
+}
+
+func BenchmarkUpgradeIdentity(b *testing.B) {
+	ident, _ := identity.Create(validConf, "userID")
+	for i := 0 ; i < b.N ; i++ {
+		identity.UpgradeIdentity(*ident)
```

Comment:
Indeed Go will measure the time for the whole function and not only the "interesting" loop. I'd say this is not important since the benchmarks are run many times and we mainly want to compare the benchmarks knowing that `Create` will not have a significant impact on the overall results. I can wrap that in a `b.Run` if you prefer.